src/DID/DIDDocument.ts

```diff
@@ -30,9 +30,9 @@ export class DIDDocument {
                 if(!latestDIDDocument) {
                     resolve();
                 }
-                let JSONDocument : { "@context" : string[], id : string,  [key: string]: any} = JSON.parse(latestDIDDocument);
+                let JSONDocument : { "@context" : string[], id : string,  [key: string]: any} = JSON.parse(latestDIDDocument as any as string);
                 //Verify if it contains a valid JSON
-                try { JSONDocument = JSON.parse(latestDIDDocument); } catch(err) { reject("JSON Parse Error: " + err) };
+                try { JSONDocument = JSON.parse(latestDIDDocument as any as string); } catch(err) { reject("JSON Parse Error: " + err) };
```

Comment:
Is this a part of the JSON.parse API?
`as any as string`

Reply:
the second optional parameter of JSON.parse is a function
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

Reply:
@sobolev-alexey Good catch. 

This _was_ required because [ReadMamStream](https://github.com/laumair/identity.ts/blob/feat/mam.js/src/IOTA/mam.ts#L94) returned an array of custom type i.e., `IMamFetchedMessage[]`. We needed to type cast it to a string before parsing it. 

It is no longer necessary now as  [ReadMamStream](https://github.com/laumair/identity.ts/blob/feat/mam.js/src/IOTA/mam.ts#L94) already returns a string array. Will update.